lib/HLSL/HLOperationLower.cpp

```diff
@@ -4597,11 +4617,13 @@ namespace {
 // Atomic intrinsics.
 struct AtomicHelper {
   AtomicHelper(CallInst *CI, OP::OpCode op, Value *h, Type *opType = nullptr);
-  AtomicHelper(CallInst *CI, OP::OpCode op, Value *h, Value *bufIdx,
-               Value *baseOffset, Type *opType = nullptr);
+  AtomicHelper(CallInst *CI, OP::OpCode op, Value *h, unsigned coordDimensions,
+               Value *coords[], Value *baseOffset, Type *opType = nullptr);
   OP::OpCode opcode;
   Value *handle;
-  Value *addr;
+  static const unsigned kMaxCoordDimensions = 3;
```

Comment:
None of the users of this constructor will have more than one coordinate, but atomic ops can have up to 3.